### ISSUE TITLE
Choice Array, Numbers Parsed To String 

### ISSUE THREAD

**Issue by ClockerZadq** [mozilla-services/react-jsonschema-form#295]
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme).

### Description

I have an Array field (choices widget) with an enum set to [1,2,3], with enumNames set to ["fred", "wilma", "barney"]. When selecting a value from the list, I get an error "0: is not of a type(s) integer" "0: is not one of enum values: 1,2,3"

### Steps to Reproduce

Replicated this in the playground with this schema:

*SCHEMA*

```
"multipleChoicesList": {
      "type": "array",
      "title": "A multiple choices list",
      "items": {
        "type": "number",
        "enum": [
          1,
          2,
          3,
          4
        ]
      },
      "uniqueItems": true
    }
```

#### Expected behavior

The value that gets selected is returned in its original form, in this case, number.

#### Actual behavior

The value that gets selected is parsed to a string.

### Version

v0.33.3

**Comment by mplis-jetsetter** [comment]
Is `choice` a custom widget that you've built?

**Comment by ClockerZadq** [comment]
@mplis-jetsetter no i wrote it to indicate the default widget. I'll remove it to prevent confusion

**Comment by oreon** [comment]
I need this behavior as well we have integer Ids and its hard to parse back and forth between strings

**Comment by mplis-jetsetter** [comment]
Do you need the ability to select multiple numbers for your list? If not, I believe you can get the behavior you want by looking at the "Number enum" field on the Numbers tab in the [playground](https://mozilla-services.github.io/react-jsonschema-form/). If you do need that behavior, it does look like there's a bug with [this](https://github.com/mozilla-services/react-jsonschema-form/blob/master/src/components/widgets/SelectWidget.js#L10) function inside SelectWidget.js.

**Comment by oreon** [comment]
Thanks jetsetter for pointing me to the right function

/**
 * This is a silly limitation in the DOM where option change event values are
 * always retrieved as strings.
 */
function processValue(type, value) {
  if (type === "boolean") {
    return value === "true";
  } else if (type === "number") {
    return asNumber(value);
  }
  return value;
}


Here is my schema , the problem is with "categorys" which is a many to many field .


   this.productSchema = {
            title: "Product",
            type: "object",
            required: ['name', 'price'
            ],
            properties: {
                name: { type: "string", title: "Name", },
                price: { type: "number", title: "Price", },

                categorys: {
                    type: "array", title: "Categorys",
                    "items": {
                        'enum': LookupService.getCategorys().map(x => x.id + ""),
                         'enumNames': LookupService.getCategorys().map(x => x.displayName)
                    }, "uniqueItems": true
                },
                displayTill: { type: "string", title: "Display Till", "format": "date", },
            }
        };

**Comment by oreon** [comment]
The thing is the if clause takes into account boolean and number but what about array of numbers ? Or maybe my schema is missing the type:"number" somewhere when I try adding type:"number inside items I get these errors

instance.categorys[0] is not of a type(s) number
instance.categorys[0] is not one of enum values: 1,2,3,4
instance.categorys[1] is not of a type(s) number
instance.categorys[1] is not one of enum values: 1,2,3,4
instance.categorys[2] is not of a type(s) number
instance.categorys[2] is not one of enum values: 1,2,3,4

**Comment by mplis-jetsetter** [comment]
I'm working on a quick PR right now that I think will fix this. You're right that `processValue` doesn't handle an array of numbers properly. All `value` attributes in an HTML `select` are strings, so the `value` passed to `processValue` is an array of strings. The `type` passed to it is `'array'`, so it just passes that value right back.

**Comment by oreon** [comment]
Thanks a lot , much appreciated.

**Comment by mplis-jetsetter** [comment]
@oremj I submitted PR #298 that I think fixes this issue. Unfortunately it looks like @n1k0 is [off for about another week](https://github.com/mozilla-services/react-jsonschema-form/pull/288#issuecomment-234610615), so it can't get merged in until then.

**Comment by oreon** [comment]
Thanks , will wait a week , this project has amazing response time for an
open source project - kudos to you and team

**Comment by n1k0** [comment]
I'm back and just reviewed the PR :) :+1:

**Comment by oreon** [comment]
Thanks a lot nicolas

**Comment by n1k0** [comment]
Fixed by #298.

**Comment by wasabigeek** [comment]
In case anyone gets stuck like I did, you will need to add a `type` property to the `items` object. Adapting @oreon's example in https://github.com/mozilla-services/react-jsonschema-form/issues/295#issuecomment-237862218:

```javascript
this.productSchema = {
            ...
            properties: {
                ...
                categorys: {
                    type: "array", title: "Categorys",
                    "items": {
                        'enum': LookupService.getCategorys().map(x => x.id + ""),
                        'enumNames': LookupService.getCategorys().map(x => x.displayName),
                        'type': 'integer'  // add a type property of "number"/"integer"
                    }, "uniqueItems": true
                },
                ...
            }
        };
```
